### PR TITLE
feat: add support for netID provider 

### PR DESCRIPTION
**Pull request by piotrmsc**
## Related issue(s)
Added support for NetID as a separate provider in Kratos config.
As discussed with @aeneasr added also screen record of the working flow.
I have also docs https://github.com/ory/docs/pull/755 describing how to set it up. 

Note: due to tech limitations of NetID redirect URIs for local testing I had to use https and fake domain `test.local` on my machine - which complicated a bit the local setup but that's a different stOry ;-) 

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

## Further Comments

https://user-images.githubusercontent.com/17701416/163158221-150c9c74-7696-4b7a-b7c3-1967e9230e80.mp4

### CODE REVIEW

File: selfservice/strategy/oidc/provider_netid.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	var claims Claims
+	if err := json.NewDecoder(resp.Body).Decode(&claims); err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	return &claims, nil
+}
+
+func (n *ProviderNetID) endpoint() (*url.URL, error) {
+	return url.Parse(defaultBrokerEndpoint)
```
Comment (aeneasr) [inline selfservice/strategy/oidc/provider_netid.go:99]: Given that this is a static value, you can also just declare it as a struct:

```
netIDEndpoint = &url.URL{Protocol: "...", "Host": "...", ...}
```

and save some error checking.

File: selfservice/strategy/oidc/provider_netid.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+func (n *ProviderNetID) oAuth2(ctx context.Context) (*oauth2.Config, error) {
+	endpoint, err := n.endpoint()
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	authUrl := *endpoint
+	tokenUrl := *endpoint
+
+	authUrl.Path = path.Join(authUrl.Path, "/authorize")
```
Comment (aeneasr) [inline selfservice/strategy/oidc/provider_netid.go:50]: You can also use `ory/x/urlx.CopyWithPath` (I think or maybe it's `AppendPath`) - just make sure it returns a copy of the struct!

**Review by aeneasr — CHANGES_REQUESTED** [review]
Awesome, thank you for your contribution! This looks pretty good and I have some ideas how to improve it further :)

File: selfservice/strategy/oidc/provider_netid.go (hunk 1 above)
Comment (piotrmsc) [reply selfservice/strategy/oidc/provider_netid.go]: Ok, I see :)

File: selfservice/strategy/oidc/provider_netid.go (hunk 2 above)
Comment (piotrmsc) [reply selfservice/strategy/oidc/provider_netid.go]: ok, thanks for the hint :)

**Comment by piotrmsc** [comment]
@aeneasr I have added changes according to your comments, I also rebased with master, could you take a look?

**Review by aeneasr — APPROVED** [review]

**Pull request merged**